subsys/bluetooth/host/gatt.c

```diff
@@ -1833,6 +1833,14 @@ ssize_t bt_gatt_attr_read(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
 	}
 
+	if ((value_len != 0U && value == NULL) || (value_len == 0U && value != NULL)) {
```

Comment:
The second restriction `(value_len == 0U && value != NULL)` seems bad to me. For variable length fields, it's convenient for the user of `bt_gatt_attr_read` to supply a valid pointer while also saying there is zero valid data behind it.

Reply:
That is a good point. Will update

Reply:
Put differently, `bt_gatt_attr_read` is a kind of `memcpy`, and `memcpy(dst, src, 0)` is legal.